Fix #2546: Rename OngoingStory to PromotedStory

adhiamboperes opened the pull request:
## Explanation

Fixes #2546 

Rename instances of ../recentlyplayed/OngoingStory to PromotedStory.

- [x] ongoingStoryListSummaryResultLiveData => promotedStoryListSummaryResultLiveData
- [x] fun subscribeToOngoingStoryList => fun subscribeToPromotedStoryList in RecentlyPlayedFragmentPresenter
- [x] fun subscribeToOngoingStoryList => fun subscribeToPromotedStoryList in RecentlyPlayedFragment
- [x] class name OngoingStoryViewModel => PromotedStoryViewModel
- [x] class name OngoingListAdapter => PromotedStoryListAdapter
- [x] private class OngoingStoryViewHolder => PromotedStoryViewHolder
- [x] interface name OngoingStoryClickListener => PromotedStoryClickListener 

Note: `ongoing_story_card`  is used to show stories in the "recently played screen", under "Played within the last week and recommended stories", shown when a user clicks "View all promoted stories" from the home screen. I have renamed it to `recently_played_story_card.xml`.

`promoted_story_card.xml` is the layout file used to show promoted stories on the home page.

I removed `PromotedStoryClickListener`, `PromotedStoryListAdapter` and `PromotedStoryViewModel` from Kdoc excemption since the kdocs have been added.

## Essential Checklist
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

adhiamboperes committed:
Rename recentlyplayed/OngoingStoryViewModel

Renamed to PromotedStoryViewModel alongside usages

adhiamboperes committed:
Rename ongoingStoryListSummaryResultLiveData

Rename to promotedStoryListSummaryResultLiveData

adhiamboperes committed:
Rename subscribeToOngoingStoryList

Rename to subscribeToPromotedStoryList

adhiamboperes committed:
Rename OngoingListAdapter

Rename to PromotedStoryListAdapter

adhiamboperes committed:
Remove unused field in PromotedStoryListAdapter

The activity previously passed into the adapter constructor had no usages.

adhiamboperes committed:
Rename instances of ongoingStory to promotedStory

adhiamboperes committed:
Rename OngoingStoryClickListener

Rename to PromotedStoryClickListener

adhiamboperes committed:
Rename OngoingStoryViewHolder to PromotedStoryViewHolder

adhiamboperes committed:
Merge branch 'develop' into rename-ongoing-story

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryClickListener.kt:
```diff
@@ -0,0 +1,8 @@
+
+/** Listener interface for when promoted story is clicked in the UI. */
+interface PromotedStoryClickListener {
+  fun promotedStoryClicked(promotedStory: PromotedStory)
```
Could we go ahead and add  KDoc here? It'll help us avoid needing to carry over the KDoc exemption.

adhiamboperes replied:
> Thanks @adhiamboperes just had a few comments.
> 
> Regarding the layouts, when is ongoing_story_card vs. promoted_story_card used? It'd be nice to fix this last case, too, if it's the only other occurrence of "ongoing story" left.

Note: `ongoing_story_card`  is used to show stories in the "recently played screen", under "Played within the last week and recommended stories", shown when a user clicks "View all promoted stories" from the home screen. I have renamed it to `recently_played_story_card.xml`.

`promoted_story_card.xml` is the layout file used to show promoted stories on the home page.

adhiamboperes replied:
I've added kdoc for the interface method

BenHenning commented on scripts/assets/kdoc_validity_exemptions.textproto:
```diff
@@ -92,9 +92,9 @@ exempted_file_path: "app/src/main/java/org/oppia/android/app/home/RouteToTopicPl
-exempted_file_path: "app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingListAdapter.kt"
-exempted_file_path: "app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryClickListener.kt"
-exempted_file_path: "app/src/main/java/org/oppia/android/app/home/recentlyplayed/OngoingStoryViewModel.kt"
+exempted_file_path: "app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryClickListener.kt"
+exempted_file_path: "app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryListAdapter.kt"
+exempted_file_path: "app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryViewModel.kt"
```
I think that it'd be nice, while here, to go ahead and finish up the KDocs for these files so that we don't need to continue carrying the exemptions.

adhiamboperes replied:
I have added the KDocs and removed these excemptions.

BenHenning reviewed (changes requested):
Thanks @adhiamboperes just had a few comments.

Regarding the layouts, when is ongoing_story_card vs. promoted_story_card used? It'd be nice to fix this last case, too, if it's the only other occurrence of "ongoing story" left.

adhiamboperes committed:
Add kdocs for promoted story files

Also removed the kdoc exemptions for the same

adhiamboperes committed:
Rename ongoing_story_card to recently_played_story_card

Note: This is the layout file used to show stories in the "recently played screen", under "Played within the last week and recommended stories"

adhiamboperes committed:
Merge branch 'rename-ongoing-story' of github.com:adhiamboperes/oppia-android into rename-ongoing-story

adhiamboperes reviewed (commented):
> Thanks @adhiamboperes just had a few comments.
> 
> Regarding the layouts, when is ongoing_story_card vs. promoted_story_card used? It'd be nice to fix this last case, too, if it's the only other occurrence of "ongoing story" left.

Note: `ongoing_story_card`  is used to show stories in the "recently played screen", under "Played within the last week and recommended stories", shown when a user clicks "View all promoted stories" from the home screen. I have renamed it to `recently_played_story_card.xml`.

`promoted_story_card.xml` is the layout file used to show promoted stories on the home page.

adhiamboperes committed:
Add kdocs for public methods and fields

Add documentation

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryClickListener.kt:
```diff
@@ -4,5 +4,9 @@ import org.oppia.android.app.model.PromotedStory
 
 /** Listener interface for when promoted story is clicked in the UI. */
 interface PromotedStoryClickListener {
+  /** Called when the a promoted story card is clicked by the user.
```
There should be a newline before the text here (to match the formatting shown in: https://developer.android.com/kotlin/style-guide#formatting_2).

Ditto elsewhere.

adhiamboperes replied:
Corrected this for all 3 files

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryClickListener.kt:
```diff
@@ -4,5 +4,9 @@ import org.oppia.android.app.model.PromotedStory
 interface PromotedStoryClickListener {
+  /** Called when the a promoted story card is clicked by the user.
+   *
+   * @param promotedStory object of the story selected
```
Perhaps: ``the [PromotedStory] that was clicked``?

adhiamboperes replied:
Yeah this reads better.

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryClickListener.kt:
```diff
@@ -4,5 +4,9 @@ import org.oppia.android.app.model.PromotedStory
+  /** Called when the a promoted story card is clicked by the user.
+   *
+   * @param promotedStory object of the story selected
+   * */
```
Should be ``*/`` (per the link above). Ditto elsewhere.

adhiamboperes replied:
Corrected everywhere

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryListAdapter.kt:
```diff
@@ -3,13 +3,16 @@ package org.oppia.android.app.home.recentlyplayed
-/** Adapter to inflate different items/views inside [RecyclerView] for Ongoing Story List. */
+/** Adapter to inflate different items/views inside [RecyclerView] for Ongoing Story List.
+ *
+ * @param [itemList] list of items that may be displayed in recently-played fragment recycler view.
```
```suggestion
 * @property itemList list of items that may be displayed in recently-played fragment recycler view
```

Three things:
1. We use 'property' instead of 'param' for constructor vals.
2. We should reference properties and parameters directly by name rather than using link syntax.
1. The first sentence fragment for properties and parameters don't end in punctuation.

Ditto elsewhere in the PR (for all three).

adhiamboperes replied:
Just to clarify that you meant in (3) that we don't punctuate if there is only one line for properties and parameters?

I've made the changes.

BenHenning replied:
Thanks! And your interpretation is correct. We only punctuate when there's more than one sentence for parameters and properties, otherwise we leave it off.

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryListAdapter.kt:
```diff
@@ -76,21 +79,29 @@ class PromotedStoryListAdapter(
   }
 
+  /**
+   *  Defines the number of columns of recently played stories shown in the recently played stories
+   *  list.
```
Suggest s/Defines/Specifies/

Also, I think you have an extra space before the lines in this KDoc.

adhiamboperes replied:
I have gone with Specifies, and also removed the extra space.

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryListAdapter.kt:
```diff
@@ -76,21 +79,29 @@ class PromotedStoryListAdapter(
   private class SectionTitleViewHolder(
     val binding: SectionTitleBinding
   ) : RecyclerView.ViewHolder(binding.root) {
+    /** Binds the viewmodel that sets section titles */
```
```suggestion
    /** Binds the view model that sets section titles. */
```

Summary fragments do end in punctuation, and 'view model' is two words.

Ditto elsewhere.

adhiamboperes replied:
Noted and fixed.

BenHenning reviewed (commented):
Thanks @adhiamboperes! I think the content for the new docs is generally good, but there are a few formatting issues to address for consistency and per the style guide. PTAL.

adhiamboperes committed:
Rectify KDoc formatting

adhiamboperes reviewed (commented):
I have implemented the KDoc formatting suggestions, PTAL @BenHenning

adhiamboperes committed:
Merge branch 'develop' into rename-ongoing-story

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryViewModel.kt:
```diff
@@ -124,9 +124,11 @@ class PromotedStoryViewModel(
-  /** Creates the content description of the story thumbnail by interpolating
-   * the thumbnail description string resource and the next chapter title
-   * */
+  /**
+   * Creates the content description of the story thumbnail by interpolating the thumbnail
+   * description string resource and the next chapter title.
+   *
+   */
```
```suggestion
  /**
   * Creates the content description of the story thumbnail by interpolating the thumbnail
   * description string resource and the next chapter title.
   */
```

BenHenning replied:
This appears to be resolved in the latest changes.

BenHenning commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/PromotedStoryClickListener.kt:
```diff
@@ -4,9 +4,10 @@ import org.oppia.android.app.model.PromotedStory
 interface PromotedStoryClickListener {
-  /** Called when the a promoted story card is clicked by the user.
+  /**
+   * Called when a promoted story card is clicked by the user
```
```suggestion
   * Called when a promoted story card is clicked by the user.
```

BenHenning replied:
This appears to be resolved in the latest changes.

BenHenning reviewed (commented):
Thanks @adhiamboperes! Just 2 small nits, otherwise the PR LGTM.

adhiamboperes committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into rename-ongoing-story

adhiamboperes committed:
Merge branch 'rename-ongoing-story' of github.com:adhiamboperes/oppia-android into rename-ongoing-story

adhiamboperes committed:
Correct some more KDoc formatting

adhiamboperes commented:
@BenHenning I'm not sure why the workflows did not run. Could you PTAL?

BenHenning reviewed (approved):
Thanks @adhiamboperes! PR LGTM.

BenHenning commented:
@adhiamboperes they sometimes hang, though it's a bit rare. I'll update the PR to the latest develop since it needs it, anyway, and that should retrigger a run.

@rt4914 PTAL for codeowners.

BenHenning committed:
Merge branch 'develop' into rename-ongoing-story

rt4914 commented on app/src/main/res/layout/recently_played_story_card.xml:
```diff
@@ -8,7 +8,7 @@
+      type="org.oppia.android.app.home.recentlyplayed.PromotedStoryViewModel" />
   </data>
 
   <com.google.android.material.card.MaterialCardView
```
@adhiamboperes Please change the id name of `ongoing_story_card_view ` to `promoted_story_card_view `

rt4914 replied:
Also in line `31` we are using `clickOnOngoingStoryTile` which should also be renamed to `clickOnPromotedStoryTile`

adhiamboperes replied:
Thanks @rt4914, I have updated these, PTAL.

adhiamboperes replied:
Thanks @rt4914, I have updated these.

rt4914 reviewed (changes requested):
@adhiamboperes Suggested minor changes. Thanks.

adhiamboperes committed:
Rename leftover ongoing story instances.

Change the id name of ongoing_story_card_view to promoted_story_card_view in recently_played_story_card.xml.
Also renamed clickOnOngoingStoryTile to clickOnPromotedStoryTile in PromotedStoryViewModel.

adhiamboperes reviewed (commented):
@rt4914, PTAL again.

rt4914 reviewed (approved):
LGTM

adhiamboperes committed:
Merge branch 'develop' into rename-ongoing-story

BenHenning reviewed (approved):
Latest changes LGTM, thanks @adhiamboperes!
